internal/notification/discord.go

```diff
@@ -0,0 +1,196 @@
+// Copyright (c) 2021 - 2024, Ludvig Lundgren and the autobrr contributors.
+// Code is heavily modified for use with seasonpackarr
+// SPDX-License-Identifier: GPL-2.0-or-later
+
+package notification
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"strings"
+	"time"
+
+	"seasonpackarr/internal/config"
+	"seasonpackarr/internal/domain"
+	"seasonpackarr/internal/logger"
+	"seasonpackarr/pkg/errors"
+
+	"github.com/rs/zerolog"
+)
+
+type DiscordMessage struct {
+	Content interface{}     `json:"content"`
+	Embeds  []DiscordEmbeds `json:"embeds,omitempty"`
+}
+
+type DiscordEmbeds struct {
+	Title       string                `json:"title"`
+	Description string                `json:"description"`
+	Color       int                   `json:"color"`
+	Fields      []DiscordEmbedsFields `json:"fields,omitempty"`
+	Timestamp   time.Time             `json:"timestamp"`
+}
+
+type DiscordEmbedsFields struct {
+	Name   string `json:"name"`
+	Value  string `json:"value"`
+	Inline bool   `json:"inline,omitempty"`
+}
+
+type EmbedColors int
+
+const (
+	LIGHT_BLUE EmbedColors = 5814783  // 58b9ff
+	RED        EmbedColors = 15548997 // ed4245
+	GREEN      EmbedColors = 5763719  // 57f287
+	GRAY       EmbedColors = 10070709 // 99aab5
+)
+
+type DiscordSender struct {
+	log zerolog.Logger
+	cfg *config.AppConfig
+
+	httpClient *http.Client
+}
+
+func NewDiscordSender(log logger.Logger, config *config.AppConfig) *DiscordSender {
+	return &DiscordSender{
+		log: log.With().Str("sender", "discord").Logger(),
+		cfg: config,
+		httpClient: &http.Client{
+			Timeout: time.Second * 30,
+		},
+	}
+}
+
+func (s *DiscordSender) Send(statusCode int, payload domain.NotificationPayload) error {
+	if !s.isEnabled() {
+		return nil
+	}
+
+	m := DiscordMessage{
+		Content: nil,
+		Embeds:  []DiscordEmbeds{s.buildEmbed(statusCode, payload)},
+	}
+
+	jsonData, err := json.Marshal(m)
+	if err != nil {
+		s.log.Error().Err(err).Msgf("discord client could not marshal data: %v", m)
```

Comment:
You already have logging on the caller side in your processor. With this you'll get double logs. Maybe you want it or not. I would just return wrapped errors here and log them in your handler like you already do.